### PR TITLE
Update to generator 5.2.1

### PR DESCRIPTION
**Pull request by pcothenet**
### What

- Update to latest version of [https://github.com/OpenAPITools/openapi-generator](https://github.com/OpenAPITools/openapi-generator)
- Set the UserAgent to a custom version

Generated by https://github.com/patch-technology/client-code-generation/pull/30

### Why

- Better track each version of the SDK
- Bring in bug fixes and performance improvements

### SDK Release Checklist

- [x] Have you added an integration test for the changes?
- [x] Have you built the gem locally and made queries against it successfully?
- [x] Did you update the changelog?
- [x] Did you bump the package version?
- [ ] For breaking changes, did you plan for the release of the new SDK versions and deploy the API to production?

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,11 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [1.10.0] - 2021-08-27
+
+### Added
+
+- Adds custom User-Agent header
```
Comment (kleinjm) [inline CHANGELOG.md:12]:
```suggestion
- Adds custom User-Agent header

```

File: Gemfile [hunk 2]
```diff
@@ -5,6 +5,5 @@ gemspec
 group :development, :test do
   gem 'rake', '~> 13.0.1'
   gem 'pry-byebug'
-  gem 'rubocop'
-  gem 'factory_bot'
```
Comment (kleinjm) [inline Gemfile:9]: is factorybot a dependency of another gem? I see it's removed here but still in the lock file

File: lib/patch_ruby/api_client.rb [hunk 3]
```diff
@@ -155,6 +156,44 @@ def build_request_body(header_params, form_params, body)
...
     end
 
+    # Save response body into a file in (the defined) temporary folder, using the filename
+    # from the "Content-Disposition" header if provided, otherwise a random filename.
+    # The response body is written to the file in chunks in order to handle files which
+    # size is larger than maximum Ruby String or even larger than the maximum memory a Ruby
+    # process can use.
+    #
+    # @see Configuration#temp_folder_path
+    def download_file(request)
```
[the remark this answers is not in the document]
Comment (kleinjm) [reply lib/patch_ruby/api_client.rb:166]: It's only there to support the `if opts[:return_type] == 'File'` case. I don't think it's hurting anything and could one day be useful so I'd leave it

**Review by kleinjm — APPROVED** [review]
Not looking super closely at every file but LGTM based on how the generator is now configured. Nice work!

File: Gemfile (hunk 2 above)
Comment (pcothenet) [reply Gemfile:9]: It gets added via the `gemspec` file (interestingly the template does not manage the Gemfile directly)

**Pull request merged**